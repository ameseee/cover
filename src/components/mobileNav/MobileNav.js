import React from 'react';
import { Link } from 'react-router-dom';
//import '../../index.css';
import PropTypes from 'prop-types';

const MobileNav = ({ history }) => {

  const closeMobileNav = () => {
    history.goBack();
  };

  return (
    <section className="drop-down">
      <button
        className="go-back"
        onClick={() => closeMobileNav()}>
      </button>
      <div>
        <Link
          to="/main"
          className="drop-down-option"
        >
          Home
        </Link>
      </div>
      <div>
        <Link
          to="/contacts"
          className="drop-down-option"
        >
          Contacts
        </Link>
      </div>
      <div>
        <Link
          to="/"
          className="drop-down-option"
        >
          Sign Out
        </Link>
      </div>
    </section>
  );
};

MobileNav.propTypes = {
  history: PropTypes.object,
  showHamburger: PropTypes.func,
};

export default MobileNav;
