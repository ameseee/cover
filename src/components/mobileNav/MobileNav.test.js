import React from 'react';
import MobileNav from './MobileNav';
import { shallow } from 'enzyme';

describe('<MobileNav />', () => {
  let wrapper;
  let mockFn = jest.fn();

  beforeEach( () => {
    const wrapper = shallow(<MobileNav />);
  });

  it('should exist', () => {
    expect(wrapper).toBeDefined();
  });

  it.skip('should render a "go back" button', () => {
    const goBackBtn = wrapper.find('.go-back');

    expect(goBackBtn.length).toEqual(1);
  });

  it.skip('should render a four drop down option links', () => {
    const option = wrapper.find('.drop-down-option');

    expect(option.length).toEqual(4);
  });

  it.skip('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

});
