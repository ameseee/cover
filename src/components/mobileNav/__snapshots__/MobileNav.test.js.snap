// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MobileNav /> should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MobileNav />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button
          className="go-back"
          onClick={[Function]}
        />,
        <div>
          <Link
            className="drop-down-option"
            replace={false}
            to="/main"
          >
            Home
          </Link>
        </div>,
        <div>
          <Link
            className="drop-down-option"
            replace={false}
            to="/about"
          >
            About
          </Link>
        </div>,
        <div>
          <Link
            className="drop-down-option"
            replace={false}
            to="/contacts"
          >
            Contacts
          </Link>
        </div>,
        <div>
          <Link
            className="drop-down-option"
            replace={false}
            to="/"
          >
            Sign Out
          </Link>
        </div>,
      ],
      "className": "drop-down",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "className": "go-back",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": "button",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": <Link
            className="drop-down-option"
            replace={false}
            to="/main"
          >
            Home
          </Link>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "children": "Home",
            "className": "drop-down-option",
            "replace": false,
            "to": "/main",
          },
          "ref": null,
          "rendered": "Home",
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": <Link
            className="drop-down-option"
            replace={false}
            to="/about"
          >
            About
          </Link>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "children": "About",
            "className": "drop-down-option",
            "replace": false,
            "to": "/about",
          },
          "ref": null,
          "rendered": "About",
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": <Link
            className="drop-down-option"
            replace={false}
            to="/contacts"
          >
            Contacts
          </Link>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "children": "Contacts",
            "className": "drop-down-option",
            "replace": false,
            "to": "/contacts",
          },
          "ref": null,
          "rendered": "Contacts",
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": <Link
            className="drop-down-option"
            replace={false}
            to="/"
          >
            Sign Out
          </Link>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "children": "Sign Out",
            "className": "drop-down-option",
            "replace": false,
            "to": "/",
          },
          "ref": null,
          "rendered": "Sign Out",
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            className="go-back"
            onClick={[Function]}
          />,
          <div>
            <Link
              className="drop-down-option"
              replace={false}
              to="/main"
            >
              Home
            </Link>
          </div>,
          <div>
            <Link
              className="drop-down-option"
              replace={false}
              to="/about"
            >
              About
            </Link>
          </div>,
          <div>
            <Link
              className="drop-down-option"
              replace={false}
              to="/contacts"
            >
              Contacts
            </Link>
          </div>,
          <div>
            <Link
              className="drop-down-option"
              replace={false}
              to="/"
            >
              Sign Out
            </Link>
          </div>,
        ],
        "className": "drop-down",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "className": "go-back",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": "button",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": <Link
              className="drop-down-option"
              replace={false}
              to="/main"
            >
              Home
            </Link>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {
              "children": "Home",
              "className": "drop-down-option",
              "replace": false,
              "to": "/main",
            },
            "ref": null,
            "rendered": "Home",
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": <Link
              className="drop-down-option"
              replace={false}
              to="/about"
            >
              About
            </Link>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {
              "children": "About",
              "className": "drop-down-option",
              "replace": false,
              "to": "/about",
            },
            "ref": null,
            "rendered": "About",
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": <Link
              className="drop-down-option"
              replace={false}
              to="/contacts"
            >
              Contacts
            </Link>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {
              "children": "Contacts",
              "className": "drop-down-option",
              "replace": false,
              "to": "/contacts",
            },
            "ref": null,
            "rendered": "Contacts",
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": <Link
              className="drop-down-option"
              replace={false}
              to="/"
            >
              Sign Out
            </Link>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {
              "children": "Sign Out",
              "className": "drop-down-option",
              "replace": false,
              "to": "/",
            },
            "ref": null,
            "rendered": "Sign Out",
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
